src/util/animate.js

```diff
@@ -1,4 +1,64 @@
-(function() {
+(function () {
+
+  var extend = fabric.util.object.extend,
+      clone = fabric.util.object.clone;
+
+  /**
+   * @typedef {Object} AnimationOptions
+   * @property {Function} [options.onChange] Callback; invoked on every value change
+   * @property {Function} [options.onComplete] Callback; invoked when value change is completed
+   * @property {Number} [options.startValue=0] Starting value
+   * @property {Number} [options.endValue=100] Ending value
+   * @property {Number} [options.byValue=100] Value to modify the property by
+   * @property {Function} [options.easing] Easing function
+   * @property {Number} [options.duration=500] Duration of change (in ms)
+   * @property {Function} [options.abort] Additional function with logic. If returns true, animation aborts.
+   *
+   * @typedef {() => void} CancelFunction
+   *
+   * @typedef {(AnimationOptions & { cancel: CancelFunction }} AnimationContext
+   */
+
+  /**
+   * Array holding all running animations
+   * @memberof fabric
+   * @type {AnimationContext[]}
+   */
+  var RUNNING_ANIMATIONS = [];
+  fabric.util.object.extend(RUNNING_ANIMATIONS, {
+
+    /**
+     * cancel all running animations
+     */
+    cancelAll: function () {
+      var animations = this.splice(0, this.length);
+      animations.forEach(function (animation) {
+        animation.cancel();
+      });
+    },
+
+    /**
+     *
+     * @param {CancelFunction} cancelFunc the function returned by animate
+     * @returns {number}
+     */
+    findAnimationIndex: function (cancelFunc) {
+      return this.findIndex(function (animation) {
+        return animation.cancel === cancelFunc;
+      });
```

Comment:
we still support ie11.
Since we have a loop to find findAnimation and findIndex i think is not supported in ie11, this should be:

```
return this.indexOf(findAnimation(cancelFunc));
```

Reply:
456a36052dda7a16009849017a8122f504e81b00